Relax octokit version dependency

andriusm opened the pull request:
Previous constrain was limiting where Danger 9.x cannot be used together with Pronto in a project because of different octokit dependency version requirements. There was also an issue registered suggesting this - https://github.com/prontolabs/pronto/issues/430

ashkulz commented on lib/pronto/version.rb:
```diff
@@ -1,6 +1,6 @@
 module Pronto
   module Version
-    STRING = '0.11.0'.freeze
```
Please don't change this.

ashkulz commented on pronto.gemspec:
```diff
@@ -42,7 +42,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency('gitlab', '~> 4.4', '>= 4.4.0')
   s.add_runtime_dependency('httparty', '>= 0.13.7')
-  s.add_runtime_dependency('octokit', '~> 4.7', '>= 4.7.0')
+  s.add_runtime_dependency('octokit', '~> 5.0')
```
```suggestion
  s.add_runtime_dependency('octokit', '>= 4.7.0')
```
Maybe just remove the `~> 4.7` part?

andriusm commented:
Changed octokit dependency to be `>= 4.7.0` and removed several versions of Ruby from the build matrix which were failing. Octokit version 5.0 dropped support for Ruby versions 2.0, 2.1, 2.2, 2.3, 2.4, 2.5 and 2.6, which have all reached end-of-life.

ashkulz reviewed (changes requested):
I wouldn't like to drop support for Ruby 2.6 although it's EOL, because that's the currently supported JRuby version.

Dropping support for old Ruby versions will also be problematic, as that'll require a major version bump. Is there a way to fix the CI without doing all this?

ashkulz commented:
Any update, @andriusm?
